docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+[[analysis-synonyms]]
+== Synonyms
+
+Synonyms are words or phrases that have the same or similar meaning.
+They are an important aspect of search, as they can improve the search experience and the relevancy of search results.
+
+Synonyms allow you to:
+
+* Improve search relevance: Synonyms enable users to find relevant documents even if they use different terms to express the same concept.
+* Increase search scope: Using synonyms ensures a wider range of possible terms. Search results are not limited to the specific terms included in your documents, and variations of relevant terms are included in search results.
```

Comment:
This seems to me just a paraphrase of the previous item, we should combine them into the one.

Synonyms usually improve recall or scope which is one of the component of search relevance.

Reply:
Sounds good 👍